ui/src/datasets.tsx

```diff
@@ -81,6 +91,64 @@ export function Datasets() {
     });
   };
 
+  const listConceptSets = (
+    editable: boolean,
+    conceptSets: { id: string; name: string }[]
+  ) => {
+    return conceptSets.map((conceptSet) => (
+      <Stack key={conceptSet.name} direction="row" alignItems="center">
+        <Checkbox
+          size="small"
+          fontSize="inherit"
+          name={conceptSet.name}
+          checked={selectedConceptSets.has(conceptSet.id)}
+          onChange={() => onToggle(updateSelectedConceptSets, conceptSet.id)}
```

Comment:
This would also be an existing issue with cohorts, if it is an issue: what happens when you switch underlays? Is the redux data being reset somewhere?

Reply:
Fixed. The cohorts and concept sets are stored for all underlays and filtered depending on which is active. This was done correctly for cohorts but I forgot to do it for concept sets.